.gitlab-ci.yml

```diff
@@ -0,0 +1,23 @@
+stages:
+  - build
+
+default:
+  image: docker:24.0.7-git
+  services:
+    - docker:24.0.7-dind
+  before_script:
+    - docker info
+
+build_image:
+  stage: build
+  script:
+    - IMAGE_NAME=$(echo $CI_COMMIT_TAG | cut -d / -f 1)
+    - IMAGE_VERSION=$(echo $CI_COMMIT_TAG | cut -d / -f 2)
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+    - docker build --compress --squash --rm -t $CI_REGISTRY_IMAGE/$IMAGE_NAME:$IMAGE_VERSION -f $IMAGE_NAME/Dockerfile --build-arg BUILD_TAG=$IMAGE_VERSION --no-cache $IMAGE_NAME
+    - docker push $CI_REGISTRY_IMAGE/$IMAGE_NAME:$IMAGE_VERSION
+    - docker logout
+  tags:
+    - docker-machine
```

Comment:
Who is providing this tag? Are we sure this will be kept? (I thought the ci team was providing a temporary solution for docker builds, while people didn't upgrade to the properly supported alternative - kaniko)

Reply:
This tag is provided by the existing gitlab-runners with privileged permissions, to be able to use docker in docker. I implemented an alternative solution with kaniko but it generates images with a bigger size than the ones generated by docker.